Build failing for pull request.

On creating a pull reuqest the build fails.
Please refer #218

As seen in last commit's [build status](https://travis-ci.org/ncs-jss/HTTP_200/builds/298167524?utm_source=github_status&utm_medium=notification) #226

Travis CI build fails with following errors-
```
./notices/decorators.py:18:9: E722 do not use bare except'
./notices/decorators.py:44:9: E722 do not use bare except'
./notices/views.py:80:9: E722 do not use bare except'
./notices/views.py:146:9: E722 do not use bare except'
./notices/views.py:250:9: E722 do not use bare except'
./notices/views.py:273:9: E722 do not use bare except'
./notices/views.py:289:13: E722 do not use bare except'
./notices/views.py:297:9: E722 do not use bare except'
./notices/views.py:319:13: E722 do not use bare except'
./notices/views.py:322:9: E722 do not use bare except'
./notices/views.py:326:13: E722 do not use bare except'
./notices/views.py:330:13: E722 do not use bare except'
./notices/views.py:334:13: E722 do not use bare except'
./wifi/models.py:38:9: E722 do not use bare except'
./wifi/views.py:29:9: E722 do not use bare except'
./wifi/views.py:38:9: E722 do not use bare except'
./wifi/views.py:116:13: E722 do not use bare except'

The command "flake8 ./" exited with 1.
```

***Solution-***
After reading PEP8 documentation-

> A bare except: clause will catch SystemExit and KeyboardInterrupt exceptions, making it harder to interrupt a program with Control-C, and can disguise other problems. If you want to catch all exceptions that signal program errors, use except Exception: (bare except is equivalent to except BaseException:).

Hence all the bare ```except:``` statements must be updated with ```except BareException:```

--- a/notices/decorators.py
+++ b/notices/decorators.py
@@ -15,7 +15,7 @@ def student_profile_complete(function):
         try:
             user = User.objects.get(username=request.user.username)
             profile = StudentDetail.objects.get(user=user)
-        except:
+        except BaseException:
             return function(request, *args, **kwargs)
 
         if re.split('(\d+)', str(user))[2] == "mca" or re.split('(\d+)', str(user))[2] == "mba" or re.split('(\d+)', str(user))[2] == "MCA" or re.split('(\d+)', str(user))[2] == "MBA":
@@ -41,7 +41,7 @@ def default_password_change(function):
     def password(request, *args, **kwargs):
         try:
             user = User.objects.get(username=request.user.username)
-        except:
+        except BaseException:
             return function(request, *args, **kwargs)
 
         if check_password(str(user).lower(), user.password) or check_password(str(user).upper(), user.password):

--- a/notices/views.py
+++ b/notices/views.py
@@ -77,7 +77,7 @@ class NoticeList(LoginRequiredMixin, generic.View):
         # get objects from paginator according to page number
         try:
             notices = paginator.page(page_num)
-        except:
+        except BaseException:
             notices = paginator.page(paginator.num_pages)
 
         context_dict = {
@@ -143,7 +143,7 @@ class CreateNotice(LoginRequiredMixin, GroupRequiredMixin, CreateView):
     def dispatch(self, request, *args, **kwargs):
         try:
             FacultyDetail.objects.get(user__id=self.request.user.id)
-        except:
+        except BaseException:
             raise PermissionDenied
         return super(CreateNotice, self).dispatch(request, *args, **kwargs)
 
@@ -247,7 +247,7 @@ class BookmarkListView(LoginRequiredMixin, generic.ListView):
         # get objects from paginator according to page number
         try:
             bookmarks = paginator.page(page_num)
-        except:
+        except BaseException:
             bookmarks = paginator.page(paginator.num_pages)
 
         context_dict = {
@@ -270,7 +270,7 @@ class BookmarkDeleteView(LoginRequiredMixin, DeleteView):
             BookmarkedNotice.objects.filter(user=self.request.user, notice=notice)[0].delete()
             messages.success(self.request, 'Bookmark removed.')
             return HttpResponse("Deleted the notice Successfully")
-        except:
+        except BaseException:
             messages.success(self.request, 'Some error occured.')
             return HttpResponse("Some error occured. Please try after sometime.")
 
@@ -286,7 +286,7 @@ class PinCreateView(LoginRequiredMixin, generic.View):
                 previous = BookmarkedNotice.objects.get(user=request.user, pinned=True)
                 previous.pinned = False
                 previous.save()
-            except:
+            except BaseException:
                 pass
 
             bookmark = BookmarkedNotice.objects.get(pk=pk)
@@ -294,7 +294,7 @@ class PinCreateView(LoginRequiredMixin, generic.View):
             bookmark.save()
             return HttpResponse("Pinned To Top")
 
-        except:
+        except BaseException:
             return HttpResponse("Some error occured.Please try after sometime")
 
 
@@ -316,22 +316,22 @@ class ReleventNoticeListView(LoginRequiredMixin, generic.View):
             notices = notices.filter(Q(course_branch_year__icontains='-AllBranches-'))
             try:
                 notices = notices.filter(course_branch_year__icontains=faculty.department)
-            except:
+            except BaseException:
                 pass
 
-        except:
+        except BaseException:
             student = get_object_or_404(StudentDetail, user__id=self.request.user.id)
             try:
                 notices = notices.filter(Q(course_branch_year__icontains=student.course+"-") | Q(course_branch_year__icontains='AllCourses-'))
-            except:
+            except BaseException:
                 pass
             try:
                 notices = notices.filter(Q(course_branch_year__icontains="-"+student.branch+"-") | Q(course_branch_year__icontains='-AllBranches-'))
-            except:
+            except BaseException:
                 pass
             try:
                 notices = notices.filter(Q(course_branch_year__icontains="-"+str(student.year)+"-") | Q(course_branch_year__icontains='-AllYears-'))
-            except:
+            except BaseException:
                 pass
 
         else:

--- a/wifi/models.py
+++ b/wifi/models.py
@@ -35,7 +35,7 @@ class WifiDetail(models.Model):
             user = User.objects.get(username=self.user)
             profile = StudentDetail.objects.get(user=user)
             return "%s" % (profile.branch)
-        except:
+        except BaseException:
             user = User.objects.get(username=self.user)
             profile = FacultyDetail.objects.get(user=user)
             return "%s" % (profile.department)

--- a/wifi/views.py
+++ b/wifi/views.py
@@ -26,7 +26,7 @@ class StudentWifiForm(LoginRequiredMixin, View):
             mac_address = WifiDetail.objects.get(user=user)
             if mac_address:
                 return render(request, 'wifi/studentwifiform.html', {"user": user, "details": details, "mac_address": mac_address})
-        except:
+        except BaseException:
             return render(request, 'wifi/studentwifiform.html', {"user": user, "details": details})
 
     def post(self, request):
@@ -35,7 +35,7 @@ class StudentWifiForm(LoginRequiredMixin, View):
         try:
             detail = WifiDetail.objects.get(user=user)
             wifi_form = WifiForm(request.POST, instance=detail)
-        except:
+        except BaseException:
             wifi_form = WifiForm(request.POST)
 
         if wifi_form.is_valid():
@@ -113,7 +113,7 @@ class excel_writer(LoginRequiredMixin, View):
                 worksheet.write(row, 7, users.new_laptop_mac_address)
                 worksheet.write(row, 8, date_registered)
                 row += 1
-            except:
+            except BaseException:
                 user = User.objects.get(username=users.user)
                 faculty = FacultyDetail.objects.get(user=user)
                 worksheet.write(row, 0, user.username)

